src/components/Checkout/AddressForm.js

```diff
@@ -1,11 +1,31 @@
-import * as React from 'react';
-import Grid from '@mui/material/Grid';
-import Typography from '@mui/material/Typography';
-import TextField from '@mui/material/TextField';
-import FormControlLabel from '@mui/material/FormControlLabel';
-import Checkbox from '@mui/material/Checkbox';
+import * as React from "react";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import TextField from "@mui/material/TextField";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import InputLabel from "@mui/material/InputLabel";
+import Select from "@mui/material/Select";
+import MenuItem from "@mui/material/MenuItem";
+import countriesData from "./countries_states.json";
 
 export default function AddressForm() {
+  const [country, setCountry] = React.useState("");
+  const [regions, setRegions] = React.useState([]);
+  const [region, setRegion] = React.useState("");
+
+  const handleChangeCountry = (event) => {
+    setCountry(event.target.value);
+    const filteredCountry = countriesData.countries.filter(
+      (countryData) => countryData.country === event.target.value
+    );
+    setRegions(filteredCountry[0].states);
```

Comment:
Instead of using a `useState` for the regions, what do you think about not keeping it in the state of the component but processing it directly in the component? This way you won't need to remember to call this filtering every time the country changes.

Also, a good idea would be to move the state of the selected country and region to the parent component in a way that we're able to change the selected country from there and even get the state there if we need to pass it to some other component. This technique is known as Lift state up (https://reactjs.org/docs/lifting-state-up.html)